tests/Php/ConfigBuilderTest.php

```diff
@@ -0,0 +1,67 @@
+<?php
+/**
+ * Copyright © 2017 Maks Rafalko
+ *
+ * License: https://opensource.org/licenses/BSD-3-Clause New BSD License
+ */
+declare(strict_types=1);
+
+namespace Infection\Tests\Php;
+
+use Infection\Php\ConfigBuilder;
+use PHPUnit\Framework\TestCase;
+
+class ConfigBuilderTest extends TestCase
+{
+    public static $envOriginal;
+
+    /**
+     * @var string
+     */
+    private $workspace;
+
+    protected function setUp()
+    {
+        $this->workspace = sys_get_temp_dir() . DIRECTORY_SEPARATOR . microtime(true) . random_int(100, 999);
+        mkdir($this->workspace, 0777, true);
+    }
+
+    public function test_it_builds_return_existing_path()
+    {
+        $builder = new ConfigBuilder(sys_get_temp_dir());
+
+        $file = $this->workspace . DIRECTORY_SEPARATOR . 'infectionTest';
+
+        touch($file);
+
+        $this->setEnv($file);
+
+        $this->assertSame($file, $builder->build());
+    }
+
+    public static function setUpBeforeClass()
+    {
+        // Save current state
+        self::$envOriginal = getenv(ConfigBuilder::ENV_TEMP_PHP_CONFIG_PATH);
+    }
+
+    protected function tearDown()
```

Comment:
I think it makes more sense to have those methods in the following order:

- setUpBeforeClass
- tearDownAfterClass
- setUp
- tearDown
- test methods
- other methods

mostly because they define the behaviour/environment in which the declare test functions are going to be executed, having some at the end of the class makes it a bit harder to follow (IMO)